controllers/databaseclusterrestore_controller.go

```diff
@@ -86,7 +88,31 @@ func (r *DatabaseClusterRestoreReconciler) Reconcile(ctx context.Context, req ct
 
 	return ctrl.Result{}, nil
 }
+func (r *DatabaseClusterRestoreReconciler) ensureClusterIsReady(restore *dbaasv1.DatabaseClusterRestore) error {
+	timeoutCtx, cancel := context.WithTimeout(context.Background(), 15*time.Minute)
+	defer cancel()
+	for {
+		select {
+		case <-timeoutCtx.Done():
+			return errors.New("wait timeout exceeded")
+		default:
+			time.Sleep(time.Second)
+			cluster := &dbaasv1.DatabaseCluster{}
+			err := r.Get(context.Background(), types.NamespacedName{Name: restore.Spec.DatabaseCluster, Namespace: restore.Namespace}, cluster)
+			if err != nil {
+				return err
+			}
+			if cluster.Status.State == dbaasv1.AppStateReady {
+				return nil
+			}
+		}
+	}
```

Comment:
What happens while we are busy waiting here? Are we able to reconcile other objects or is the entire DatabaseClusterRestoreReconciler blocked?